big_tests/tests/connect_SUITE.erl

```diff
@@ -546,6 +568,45 @@ bind_server_generated_resource(Config) ->
     ?assert(is_binary(Resource)),
     ?assert(byte_size(Resource) > 0).
 
+same_resource_replaces_session(Config) ->
+    UserSpec = [{resource, <<"confict">>} | escalus_users:get_userspec(Config, alice)],
+    {ok, Alice1, _} = escalus_connection:start(UserSpec),
+
+    {ok, _Alice2, _} = escalus_connection:start(UserSpec),
+
+    ConflictError = escalus:wait_for_stanza(Alice1),
+    escalus:assert(is_stream_error, [<<"conflict">>, <<>>], ConflictError),
+    false = escalus_connection:is_connected(Alice1).
+
+clean_close_of_replaced_session(Config) ->
+    lager_ct_backend:capture(warning),
+
+    same_resource_replaces_session(Config),
+
+    lager_ct_backend:stop_capture(),
+    FilterFun = fun(_, Msg) ->
+                        re:run(Msg, "replaced_wait_timeout") /= nomatch
+                end,
+    [] = lager_ct_backend:recv(FilterFun).
+
+replaced_session_cannot_terminate(Config) ->
+    lager_ct_backend:capture(warning),
+    UserSpec = [{resource, <<"confict">>} | escalus_users:get_userspec(Config, alice)],
+    {ok, _Alice1, _} = escalus_connection:start(UserSpec),
+    [C2SPid] = children_specs_to_pids(rpc(mim(), supervisor, which_children, [ejabberd_c2s_sup])),
+    ok = rpc(mim(), sys, suspend, [C2SPid]),
+
+    {ok, _Alice2, _} = escalus_connection:start(UserSpec),
+
+    timer:sleep(2000),
+
+    rpc(mim(), sys, resume, [C2SPid]),
```

Comment:
```suggestion
    %% then...
    rpc(mim(), sys, resume, [C2SPid]),
```

I'm not sure about this one. Does this test check:
1. That the new resource disconnected because the timeout was still too short?
1. That the new resource connected because the timeout is ok thanks to configurability?

Reply:
This test verifies if the information about old session still existing after timeout is logged, so it may be debugged further.

Reply:
Got it, thanks